jekyll/_cci2/testing-orbs.md

```diff
@@ -0,0 +1,175 @@
+
+**Note** CircleCI is working to improve this limitation and welcomes feedback on your ideal mechanism for testing orbs.
+
+Another approach is to use your orb source as your `config.yml`. If you define your testing and publishing jobs in your orb, they will have access to everything defined in the orb. The downside is that you may need to add extraneous jobs and commands to your orb, and your orb will depend on any orbs that you need for testing and publishing.
+
+Yet another approach is when you run a build, publish a dev version of the orb, then do an automated push to a branch which uses a config that uses that dev version. This new build can run the tests. The downside with this approach is that the build that does the real testing is not directly tied to its commit, and you end up with multiple jobs every time you want to change your orb.
+
+### Integration Testing
+
+Sometimes you will want to test how your orbs interact with external services. There are several possible approaches depending on circumstances:
+
+* Make your orb support a dry-run functionality of whatever it is interacting with, and use that mode in your tests.
+* Do real interactions, using a properly set up test account and a separate repository that runs those tests using a published dev version of your orb.
+* Spin up a local service in another container of your job.
+
+## Orb Testing Best Practices
+
+The most significant issue when testing orbs is that it is not straightforward to push a new commit to a repository containing orb source code. This is because orbs are interpolated into an expanded `config.yml` at build inception and may not have the newest changes to the orb contained in that commit.
+
+There are several different approaches you can use to test your orbs (e.g. using inline orbs or external repositories) to ensure orb compatibility with the CircleCI platform. CircleCI is also in the process of developing newer ways for you to test your orbs. For more detailed information about these testing examples, refer to the [Emerging Testing Best Practices for Orbs](https://discuss.circleci.com/t/emerging-testing-best-practices-for-orbs/27474) page in the CircleCI Discussion Forum.
+
+## Orb Testing Methodologies
+
+### Testing Orbs Locally
+
+One of the easiest ways to test your orbs locally is to create an inline orb. Creating an inline orb enables you to test your orb while in active development. Inline orbs can be useful when developing your orb, or as a convenience for name-spacing jobs and commands in lengthy configurations.
+
+For more detailed information on how to create an inline orb, refer to the [Creating Orbs](https://circleci.com/docs/2.0/creating-orbs/#creating-inline-orbs) page in the CircleCI technical documentation.
```

Comment:
this and any other internal links to docs should be relative links

Reply:
Changed link to relative links